start_esp/start_esp.py

```diff
@@ -148,7 +170,40 @@ def start_nginx(nginx, nginx_conf):
         sys.exit(3)
 
 
-def fetch_service_config(args, service_config):
+def fetch_and_save_service_config(args, token, version, filename):
+    try:
+        # build request url
+        service_mgmt_url = SERVICE_MGMT_URL_TEMPLATE.format(args.service,
+                                                    version)
+        # Validate service config if we have service name and version
+        logging.info("Fetching the service configuration "\
+                     "from the service management service")
+        # download service config
+        config = fetch.fetch_service_json(service_mgmt_url, token)
+
+        # Save service json for ESP
+        service_config = args.config_dir + "/" + filename
+
+        try:
+            f = open(service_config, 'w+')
+            json.dump(config, f, sort_keys=True, indent=2,
+                      separators=(',', ': '))
+            f.close()
+        except IOError as err:
+            logging.error("Cannot save service config." + err.strerror)
+            sys.exit(3)
+
+    except fetch.FetchError as err:
+        logging.error(err.message)
+        sys.exit(err.code)
+
+# config_id might have invalid character for file name.
+def generate_service_config_filename(version):
+    return str(uuid.uuid5(uuid.NAMESPACE_DNS, str(version)))
```

Comment:
config_id can be set by customer. We have to escape it to avoid file creation error.